.pre-commit-config.yaml

```diff
@@ -38,7 +38,7 @@ repos:
       files: ^oteapi_dlite/.*$
 
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v1.3.0
+    rev: v1.4.1
```

Comment:
Difficult to spot possible breaking changes for `mypy`, since all version info seems to be on the commit level (?). Seems OK. Possible notable changes are:
* Don't type check most function bodies if ignoring errors (https://github.com/python/mypy/pull/14150) - will avoid `mypy` crashes caused by dependencies.
* Set soft error limit default to -1 (unlimited) (https://github.com/python/mypy/pull/15138) - by default, the full error lists will be displayed; no longer limited to 200 messages.